src/dsa/rpo_falcon512/signature.rs

```diff
@@ -182,7 +182,7 @@ fn decode_nonce(nonce: &NonceBytes) -> NonceElements {
     let mut result = [ZERO; 8];
     for (i, bytes) in nonce.chunks(5).enumerate() {
         buffer[..5].copy_from_slice(bytes);
-        result[i] = u64::from_le_bytes(buffer).into();
+        result[i] = u64::from_le_bytes(buffer).try_into().unwrap();
```

Comment:
Given that we know that the value in the buffer is guaranteed to be pretty small (i.e., only 5 bytes), I think we can use `Felt::new()` here. I'd also leave a comment explaining why this is OK.